toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1079,8 +1079,9 @@ func (g *PkgGraph) AddGoalNode(goalName string, packages []*pkgjson.PackageVer,
 		} else {
 			logger.Log.Warnf("Could not goal package %+v", pkg)
 			if strict {
-				logger.Log.Warnf("Missing %+v", pkg)
+				logger.Log.Errorf("Missing %+v", pkg)
 				err = fmt.Errorf("could not find all goal nodes with strict=true")
+				return
```

Comment:
I'll miss the lovely work around we could get by flooding the sets with enough valid hits so we were probabilistically likely to not get an error here...

(Good riddens)